src/cli/self_update.rs

```diff
@@ -630,7 +630,17 @@ fn install_bins() -> Result<()> {
     // NB: Even on Linux we can't just copy the new binary over the (running)
     // old binary; we must unlink it first.
     if rustup_path.exists() {
-        utils::remove_file("rustup-bin", &rustup_path)?;
+        if cfg!(windows) {
+            let removal_path = bin_path.join(&format!("rustup-to-delete{}", EXE_SUFFIX));
+            if removal_path.exists() {
```

Comment:
This shouldn't be needed - https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-movefileexw - pretty sure this is exposed either in the sys module or winapi - with MOVEFILE_REPLACE_EXISTING

Reply:
I wasn't certain, this sounds handy to know though.  If we decide to proceed along this path we can easily make this change.